flink-table/flink-table-common/src/main/java/org/apache/flink/table/api/dataview/ListView.java

```diff
@@ -106,7 +106,7 @@ public ListView() {
 	 * Returns an iterable of the list view.
 	 *
 	 * @throws Exception Thrown if the system cannot get data.
-	 * @return The iterable of the list or { @code null} if the list is empty.
+	 * @return The iterable of the list.
```

Comment:
This changes the API. We should return null instead, as documented. Remove the `toString` method in this class. And fix `{ @code null}` to `{@code null}`. Same applies for the MapView.

Reply:
Futhermore, make sure that all views are marked as `@PublicEvolving`. The other classes can be marked as `@Internal`.

Reply:
i think we should deprecate this method, and avoid `return null` in public method ?

Reply:
I'm also not a big supporter of returning `null` but I think the semantics are equal to Flink's `ListState.get()`:

```
The operator state value corresponding to the current input or {@code null}
	 * if the state is empty.
```

Reply:
Regarding to the `ListState.get()`, I think it maybe an outdated comment. Because all the States returned to the users will be wrapped in a `UserFacingXXX` (e.g. `UserFacingListState`). The `UserFacingXXX` will return an empty list instead of `null`.

I would prefer not return `null` and I don't think this will break user's behavior, because the null check still work.

Reply:
Yes good point. I will change the interfaces while merging.